src/software/ai/navigator/path_planner/hrvo/hrvo_simulator.cpp

```diff
@@ -0,0 +1,221 @@
+    const auto &robot_primitives = primitive_set.robot_primitives();
+    auto primitive_iter = robot_primitives.find(robot.id());
+    if (primitive_iter != robot_primitives.end()) {
+        TbotsProto::Primitive primitive = primitive_iter->second;
+        TbotsProto::Point destination_point_proto;
+
+        if (primitive.has_move()) {
+            const auto &move_primitive = primitive.move();
+            // TODO (#2418): Update implementation of Primitive to support
+            // multiple path points and remove this check
+            CHECK(move_primitive.motion_control().path().points().size() >= 2)
+                << "Empty path: "
+                << move_primitive.motion_control().path().points().size() << std::endl;
+            destination_point_proto =
+                    move_primitive.motion_control().path().points().at(1);
+            destination_point =
+                    Point(static_cast<float>(destination_point_proto.x_meters()),
+                           static_cast<float>(destination_point_proto.y_meters()));
+            speed_at_goal = move_primitive.final_speed_m_per_s();
+            max_speed = move_primitive.max_speed_m_per_s();
+        }
+    }
+
+    // Max distance which the robot can travel in one time step + scaling
+    double max_radius = (max_speed * time_step.toMilliseconds()) / 2;
+
+    RobotPath path =
+            RobotPath({PathPoint(destination_point, speed_at_goal)}, max_radius);
+
+    std::shared_ptr<HRVOAgent> agent = std::make_shared<HRVOAgent>(robot.id(), robot.currentState(),TeamSide::FRIENDLY, path,
+                                                                   ROBOT_MAX_RADIUS_METERS, max_speed, max_accel, FRIENDLY_ROBOT_RADIUS_MAX_INFLATION);
+
+    robots[robot.id()] = std::static_pointer_cast<Agent>(agent);
+
+}
+
+void HRVOSimulator::configureLVRobot(const Robot &robot) {
+    // Set goal of enemy robot to be the farthest point, when moving in the
+    // current direction
+    Segment segment(robot.position(),
+                    robot.position() + robot.velocity() * 100);
+
+    // Enemy robot should not enter the friendly defense area
+    std::unordered_set<Point> intersection_point_set =
+            intersection(world->field().friendlyDefenseArea(), segment);
+    if (intersection_point_set.empty() &&
+        contains(world->field().fieldLines(), robot.position())) {
+        // If the robot is in the field, then move in the current direction
+        // towards the field edge
+        intersection_point_set =
+                intersection(world->field().fieldLines(), segment);
+    }
+
+    if (intersection_point_set.empty()) {
+        // If there is no intersection point (robot is outside the field),
+        // continue moving in the current direction
+        intersection_point_set.insert(robot.position() +
+                                      robot.velocity() * 5);
+    }
+
+    Point destination = *intersection_point_set.begin();
```

Comment:
Lets simplify this to just setting the destination to be `robot.postion() + 2*robot.velocity()`, and add a simple comment explaining that we assume that the robot continues moving in the direction of its current velocity.